workers/tests/unittests/test_ohsome_client.py

```diff
@@ -143,11 +142,20 @@ def test_build_data_with_time(self) -> None:
                 "time": str,
             }
         )
-        layer = LayerDefinitionMock()
-        data = ohsome_client.build_data_dict(layer, self.bpolys, time="2014-01-01")
+        data = ohsome_client.build_data_dict(self.layer, self.bpolys, time="2014-01-01")
         self.assertTrue(schema.is_valid(data))
-
-        layer = LayerDefinitionMock()
-        layer.ratio_filter = "mock_ratio_filter"
+        layer = LayerDefinition(
+            name="",
+            description="",
+            endpoint="elements/length",
+            filter_="mock_filter",
+            ratio_filter="mock_ratio_filter",
+        )
         data = ohsome_client.build_data_dict(layer, self.bpolys, time="2014-01-01")
         self.assertTrue(schema.is_valid(data))
+
+    def test_query_layer_data(self):
+        data = asyncio.run(
+            ohsome_client.query(LayerData("name", "description", {"result": []}))
+        )
+        self.assertDictEqual({"result": []}, data)
```

Comment:
I think this test could be more advanced. It currently only tests if the function gives an empty result list back. This could happen by accident, even if you submitted a result to the function :wink:

Reply:
Added more tests.

Reply:
:+1: nice